go/libraries/doltcore/sqle/enginetest/dolt_server_test.go

```diff
@@ -336,3 +347,92 @@ func newConnection(t *testing.T, serverConfig sqlserver.ServerConfig) (*dbr.Conn
 	sess := conn.NewSession(nil)
 	return conn, sess
 }
+
+func TestDoltServerRunningUnixSocket(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("we define socket flag and unix socket cannot be created on Windows, so it will return error")
+	}
+
+	// TODO : socket flag is defined in this startServer method for now
+	unixSocketPath := "/tmp/mysql.sock"
+	// Running unix socket server
+	sc, serverConfig := startServer(t, false, "", unixSocketPath)
+	sc.WaitForStart()
+	require.True(t, strings.Contains(sqlserver.ConnectionString(serverConfig, "dolt"), "unix"))
+
+	t.Run("connecting mysql works", func(t *testing.T) {
```

Comment:
I think you could probably pull out this test block, since you've got unix sockets covered here and in the test block below. Let me know if I'm missing something unique that this is testing though.